channels/rdpsnd/client/pulse/rdpsnd_pulse.c

```diff
@@ -375,7 +406,7 @@ static BOOL rdpsnd_pulse_open(rdpsndDevicePlugin* device, const AUDIO_FORMAT* fo
 
 	if (pulse->latency > 0)
 	{
-		buffer_attr.maxlength = pa_usec_to_bytes(pulse->latency * 2 * 1000, &pulse->sample_spec);
+		buffer_attr.maxlength = (UINT32)-1;
```

Comment:
looks wrong, why change this?

Reply:
This is the absolute maximum number of bytes that can be stored in the buffer. We want it so large that incoming data always fits (therefore `maxlength=-1` here). If I understand correctly, remote side is feeding us appropriately sized chunks so we just need to put them into stream.

Reply:
Pulseaudio docs specify these values as `(uint32_t) -1` so that is how it is done here and in 3 (unmodified) lines below https://github.com/FreeRDP/FreeRDP/blob/master/channels/rdpsnd/client/pulse/rdpsnd_pulse.c#L380-L382

Reply:
Here is the reference from pulseaudio docs https://freedesktop.org/software/pulseaudio/doxygen/structpa__buffer__attr.html#abef20d3a6cab53f716846125353e56a4

Reply:
@i-garrison ok, but we´re supporting `C99` (and newer), so https://en.cppreference.com/w/c/types/integer is available and these casts are not required and the proper maximum of the type is used

Reply:
Technically we can use proper `c99` max values here but I was looking to keep this closer to pulseaudio docs.

Do you want me to change all of these to `c99` constants?

Reply:
I still believe this particular case could actually be an exception from general rule since it is 1) pretty well defined in `C` and 2) defined with corresponding integer cast in the 3rdparty API.

In any case, I now amended these (4) places to use `UINT32_MAX` constant, in extra commit.